.github/workflows/build.yml

```diff
@@ -0,0 +1,55 @@
+name: Build Firedrake
+
+on:
+  # Push to master or PR
+  push:
+    branches:
+      - master
+  pull_request:
+
+jobs:
+  linter:
+    name: "Build Firedrake"  
+    # The type of runner that the job will run on
+    runs-on: self-hosted
+    # The docker container to use.
+    container:
+      image: firedrakeproject/firedrake-env:latest
+    # Steps represent a sequence of tasks that will be executed as
+    # part of the jobs
+    steps:
+      - uses: actions/checkout@v2
+      - name: Cleanup
+        if: ${{ always() }}
+        run: |
+          cd ..
+          rm -rf build
```

Comment:
It seems tremendously odd that there are build artefacts left lying around inside a container between runs.

Reply:
Although https://github.community/t/how-to-properly-clean-up-self-hosted-runners/128909

Reply:
I'm not falling into that issue.

The artifacts are not left in containers between runs but in a folder on the build runner that is mounted in the containers. This is a widely known issue. I think what I have seems to be working, though it needs more testing.

Reply:
Ah ok.